test_runner/fixtures/neon_fixtures.py

```diff
@@ -542,7 +542,7 @@ class S3Storage:
     access_key: str
     secret_key: str
     endpoint: Optional[str] = None
-    prefix_in_bucket: Optional[str] = None
+    prefix_in_bucket: Optional[str] = ""
```

Comment:
Can you elaborate why this is needed?

Reply:
There were some tests that were creating files with a "/" prefix. 
My proposal is that if you want to get "/" prefixed on your path then you should have prefix="" rather than prefix=None. 

Currently this change is needed to get tests to pass, 
because before download behaved the same for "" and None but now I want None to behave differently. 
However, I think this change might not be necessary after changing `relative_path_to_s3_object` (if the tests are somehow using that function to decide where to upload files to?)

Reply:
> There were some tests that were creating files with a "/" prefix.

Do you remember which ones? Maybe we should change the tests too. 

> , I think this change might not be necessary after changing

If its not necessary then lets remove it

> if the tests are somehow using that function to decide where to upload files to?)

Python tests cannot use it directly, so in theory only real_s3 tests can reach it

Reply:
>Python tests cannot use it directly, so in theory only real_s3 tests can reach it

There is an `assert_prefix_empty()` in `test_timeline_delete.py` that does that:

```
     assert_prefix_empty(
         neon_env_builder,
         prefix="/".join(
             (
                "tenants",
                 str(env.initial_tenant),
                 "timelines",
                 str(leaf_timeline_id),
 ```

Reply:
> Python tests cannot use it directly, so in theory only real_s3 tests can reach it

I think this is not true. 
MockS3 tests basically have a "bucket" running on local host and we use all the same functions to work with them.

Reply:
> I think this is not true.

I'm confused. My comment was about tests not having access to `relative_path_to_s3_object`. They physically cant use it. Do you mean general use of prefix? Or what?

> There is an assert_prefix_empty() in test_timeline_delete.py that does that:

Dont understand your argument. Python tests still cant reach `relative_path_to_s3_object`. Join method in your example wont create path that is prefixed with "/". I e `"/".join(["a", "b"])` yields `a/b` without leading `/`.

Am I missing something? do you have an example of failing test? 

If you want to eliminate None prefix completely in python tests you can remove `Optional` from the type and always use empty string.

Reply:
I'm not sure what Anastasia's example is. 
However, IIUC when we run a regression test here is what happens:

1. python runs neon_local which in turn starts compute_ctl
2. inside compute_ctl we create an `S3Bucket` object storing a connection to an S3 Bucket
3. this `S3Bucket` object has access to a method `relative_path_to_s3_object` that it calls sometimes, e.g, when download is called

Reply:
>Dont understand your argument. Python tests still cant reach relative_path_to_s3_object. Join method in your example wont create path that is prefixed with "/". I e "/".join(["a", "b"]) yields a/b without leading /.

sorry for confusion.
my comment was unrelated and I reverted this test change.

Reply:
I've force pushed a commit so the git history is more straightforward. The only thing that happened after your last review is I added unit tests